andela/vidar-ah-backend
# #164366326 Implement user login with email

#### What does this PR do?
It implements the endpoint for users to log in using email.

#### Description of Task to be completed?
Users should be able to log in to Authors Haven by providing valid email and password and having a valid token delivered to them with which they can access other routes.

#### How should this be manually tested?
- Check .env.sample for the environment variables needed to test the application
- Run `npm run db-migrate` to run migrations
- Run `npm run build` to build the application
- Run `npm start` to start the application
- Make a POST request to `/api/v1/user/login` with a JSON object containing the following fields: `email`, `password`, both required. An error will be returned if a valid value is not provided for each.
- To stay signed in for a very long  time set a third optional parameter `remeberMe` to `true`
- If this request is successful, you will get a response with status 200, a success message, and a valid token with which to access other routes. 
- If email is not found in the system, you will get a 404 error.
- If the password does not match the corresponding email, you get a 401 error.

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
#164366326

#### Screenshots (if appropriate)
- N/A

#### Questions:
N/A
